### PR TITLE
Update for 1.18

### PR DESCRIPTION
**Pull request by dordsor21**

### CODE REVIEW

File: worldedit-bukkit/src/main/java/com/sk89q/worldedit/bukkit/adapter/BukkitImplAdapter.java
Comment (NotMyFault) [inline worldedit-bukkit/src/main/java/com/sk89q/worldedit/bukkit/adapter/BukkitImplAdapter.java:117]: How about `#fullBlock` to not use a getter here?

File: worldedit-core/src/main/java/com/fastasyncworldedit/core/function/generator/CavesGen.java
Comment (NotMyFault) [inline worldedit-core/src/main/java/com/fastasyncworldedit/core/function/generator/CavesGen.java:280]: Could be a one-liner enhanced switch

File: worldedit-core/src/main/java/com/sk89q/worldedit/extension/platform/Platform.java
Comment (NotMyFault) [inline worldedit-core/src/main/java/com/sk89q/worldedit/extension/platform/Platform.java:243]: How about dropping the getters in favor of sth like #versionMinY/MaxY?

File: worldedit-core/src/main/java/com/sk89q/worldedit/internal/Constants.java
Comment (NotMyFault) [inline worldedit-core/src/main/java/com/sk89q/worldedit/internal/Constants.java:47]: Can we even deprecate it for removal? Afaik it's needed for a variety of schematic shenanigans for 1.13 created schematics

File: worldedit-core/src/main/java/com/sk89q/worldedit/world/block/BlockTypesCache.java
Comment (NotMyFault) [inline worldedit-core/src/main/java/com/sk89q/worldedit/world/block/BlockTypesCache.java:260]: Is it likely that this happens? We could wrap that into a try/catch and fire the exception to obtain the stacktrace to see _if_ it's shaded

File: worldedit-core/src/main/java/com/fastasyncworldedit/core/queue/IBlocks.java
Comment (NotMyFault) [inline worldedit-core/src/main/java/com/fastasyncworldedit/core/queue/IBlocks.java:142]: Can be minified into a one-liner enhanced switch

File: worldedit-core/src/main/java/com/fastasyncworldedit/core/queue/IBlocks.java
Comment (NotMyFault) [inline worldedit-core/src/main/java/com/fastasyncworldedit/core/queue/IBlocks.java:118]: Can be minified into a one-liner enhanced switch

**Review by NotMyFault — CHANGES_REQUESTED** [review]

File: worldedit-bukkit/adapters/adapter-1_18/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_18_R1/nbt/PaperweightLazyCompoundTag.java
Comment (dordsor21) [reply worldedit-bukkit/adapters/adapter-1_18/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_18_R1/nbt/PaperweightLazyCompoundTag.java:112]: Converting the whole thing to CBT would require a lot of overhaul. Quite outside of the scope of this PR, and since upstream isn't moving heavily towards CBT, we shouldn't do that either too much.

File: worldedit-bukkit/adapters/adapter-1_18/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_18_R1/nbt/PaperweightLazyCompoundTag.java
Comment (dordsor21) [reply worldedit-bukkit/adapters/adapter-1_18/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_18_R1/nbt/PaperweightLazyCompoundTag.java:99]: ? All it would do is make the if statement longer. No need for that at all

File: worldedit-bukkit/adapters/adapter-1_18/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_18_R1/nbt/PaperweightLazyCompoundTag.java
Comment (NotMyFault) [reply worldedit-bukkit/adapters/adapter-1_18/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_18_R1/nbt/PaperweightLazyCompoundTag.java:112]: Can remove that then

File: worldedit-bukkit/adapters/adapter-1_18/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_18_R1/PaperweightBlockMaterial.java
Comment (dordsor21) [reply worldedit-bukkit/adapters/adapter-1_18/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_18_R1/PaperweightBlockMaterial.java:29]: Would required LazyCompoundTag to be BinaryTag. See above/previous comment regarding this

File: worldedit-bukkit/adapters/adapter-1_18/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_18_R1/nbt/PaperweightLazyCompoundTag.java
Comment (NotMyFault) [reply worldedit-bukkit/adapters/adapter-1_18/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_18_R1/nbt/PaperweightLazyCompoundTag.java:99]: It would make the if statement longer by 6 characters in favor of removing a line

Comment (dordsor21) [reply worldedit-bukkit/adapters/adapter-1_18/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_18_R1/nbt/PaperweightLazyCompoundTag.java:99]: Ah, misinterpreted what you meant

File: worldedit-bukkit/adapters/adapter-1_18/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_18_R1/PaperweightGetBlocks.java
Comment (dordsor21) [reply worldedit-bukkit/adapters/adapter-1_18/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_18_R1/PaperweightGetBlocks.java:317]: No. Would require a large amount of refactoring. The method ultimately uses CBT anyway.

File: worldedit-bukkit/adapters/adapter-1_18/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_18_R1/PaperweightGetBlocks.java
Comment (dordsor21) [reply worldedit-bukkit/adapters/adapter-1_18/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_18_R1/PaperweightGetBlocks.java:794]: For the whole method? Would just be a comment above the line otherwise

File: worldedit-bukkit/adapters/adapter-1_18/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_18_R1/PaperweightGetBlocks.java
Comment (dordsor21) [reply worldedit-bukkit/adapters/adapter-1_18/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_18_R1/PaperweightGetBlocks.java:1092]: this is now incorrect

File: worldedit-bukkit/adapters/adapter-1_18/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_18_R1/PaperweightGetBlocks.java
Comment (NotMyFault) [reply worldedit-bukkit/adapters/adapter-1_18/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_18_R1/PaperweightGetBlocks.java:794]: Yeah, IJ complains the cast is redundant and tends to remove it if you shuffle around lines. May be good to put a noinspection on it, it's actually needed.

File: worldedit-bukkit/adapters/adapter-1_18/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_18_R1/PaperweightGetBlocks_Copy.java
Comment (dordsor21) [reply worldedit-bukkit/adapters/adapter-1_18/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_18_R1/PaperweightGetBlocks_Copy.java:71]: It's fine as is and the comment isn't needed?

File: worldedit-bukkit/adapters/adapter-1_17_1/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_17_R1_2/PaperweightFaweWorldNativeAccess.java
Comment (NotMyFault) [inline worldedit-bukkit/adapters/adapter-1_17_1/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_17_R1_2/PaperweightFaweWorldNativeAccess.java:278]: Records are always implicitly static and final

File: worldedit-bukkit/src/main/java/com/sk89q/worldedit/bukkit/adapter/BukkitImplAdapter.java
Comment (dordsor21) [reply worldedit-bukkit/src/main/java/com/sk89q/worldedit/bukkit/adapter/BukkitImplAdapter.java:117]: This is from upstream

**Review by NotMyFault — APPROVED** [review]
Looks good and is in a state to be merged. Possible nits can be addressed later.
Just let me know before merging than I can make a clean cut of release drafter.

**Comment by NotMyFault** [comment]
@SirYwell Do you want to give it a review as well? The diff is quite big but the overall merge blockers have been addressed so far.

**Comment by SirYwell** [comment]
I won't have the time to fully review this diff

**Comment by NotMyFault** [comment]
I'm heading forward to merge this tomorrow if there's no blocking feedback by then.

**Pull request merged**
